File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        std::unordered_map<int,int> mp;
        int n = nums.size();

        std::vector<int> ans;
        for(int i = 0;i < n;i++){
            if(mp.count(target - nums[i]) > 0){
                ans.push_back(i);
                ans.push_back(mp[target-nums[i]]);
            }

            mp[nums[i]] = i;
        }
        return ans;

    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    const numMap = new Map()

    // check if difference of target - nums[i] exists in map
    // if it does, we found a match => [i, numMap.get(target-nums[i])]
    // if it doesn't, add it to map

    for(let i = 0;i < nums.length;i++){

        let difference = target - nums[i]

        console.log(difference)

        if(numMap.get(difference) !== undefined){
            return [i, numMap.get(difference)]
        }

        numMap.set(nums[i], i)

    }



};